- Description: dasd: Enhanced quality of DASD statistics.
  Symptom:     Less precise statistics in some rare ERP conditions.
  Problem:     Buildclock was not filled for some ERP requests.
  Solution:    Fill the buildclock.
  Problem-ID:  9793
  Archs:       s390-31, s390-64

--- linux-2.5/drivers/s390/block/dasd_3990_erp.c	2 Jul 2004 14:01:48 -0000	1.29.2.3
+++ linux-2.5/drivers/s390/block/dasd_3990_erp.c	2 Jul 2004 14:06:29 -0000	1.29.2.4
@@ -5,7 +5,7 @@
  * Bugreports.to..: <Linux390@de.ibm.com>
  * (C) IBM Corporation, IBM Deutschland Entwicklung GmbH, 2000, 2001
  *
- * $Revision: 1.29.2.3 $
+ * $Revision: 1.29.2.4 $
  */
 
 #include <linux/timer.h>
@@ -443,6 +443,10 @@
 	/* interrupt (this enables easier enqueing of the cqr)	    */
 	if (erp->function != dasd_3990_erp_action_4) {
 
+		DEV_MESSAGE(KERN_INFO, device, 
+			    "dasd_3990_erp_action_4: first time retry"
+			    "%s", " ");
+		
 		erp->retries = 256;
 		erp->function = dasd_3990_erp_action_4;
 
@@ -1763,7 +1767,7 @@
 	erp->magic = default_erp->magic;
 	erp->expires = 0;
 	erp->retries = 256;
-	cqr->buildclk = get_clock();
+	erp->buildclk = get_clock();
 	erp->status = DASD_CQR_FILLED;
 
 	/* remove the default erp */
@@ -2306,6 +2310,7 @@
 	erp->magic    = cqr->magic;
 	erp->expires  = 0;
 	erp->retries  = 256;
+	erp->buildclk = get_clock();
 
 	erp->status = DASD_CQR_FILLED;
 

